src/flow_solver/flow_solver_cases/non_periodic_cube_flow.cpp

```diff
@@ -28,11 +57,72 @@ std::shared_ptr<Triangulation> NonPeriodicCubeFlow<dim,nstate>::generate_grid()
 template <int dim, int nstate>
 void NonPeriodicCubeFlow<dim,nstate>::display_additional_flow_case_specific_parameters() const
 {
-    // Do nothing for now.
+    this->pcout << "- - Courant-Friedrichs-Lewy number: " << this->all_param.flow_solver_param.courant_friedrichs_lewy_number << std::endl;
+}
+
+template <int dim, int nstate>
+double NonPeriodicCubeFlow<dim,nstate>::get_adaptive_time_step(std::shared_ptr<DGBase<dim,double>> dg) const
+{
+    // compute time step based on advection speed (i.e. maximum local wave speed)
+    const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+    const double approximate_grid_spacing = (this->all_param.flow_solver_param.grid_right_bound-this->all_param.flow_solver_param.grid_left_bound)/pow(number_of_degrees_of_freedom_per_state,(1.0/dim));
+    const double cfl_number = this->all_param.flow_solver_param.courant_friedrichs_lewy_number;
+    const double time_step = cfl_number * approximate_grid_spacing / this->maximum_local_wave_speed;
+    
+    return time_step;
+}
+
+template <int dim, int nstate>
+double NonPeriodicCubeFlow<dim,nstate>::get_adaptive_time_step_initial(std::shared_ptr<DGBase<dim,double>> dg)
+{
+    // initialize the maximum local wave speed
+    update_maximum_local_wave_speed(*dg);
+    // compute time step based on advection speed (i.e. maximum local wave speed)
+    const double time_step = get_adaptive_time_step(dg);
+    return time_step;
+}
+
+template<int dim, int nstate>
+void NonPeriodicCubeFlow<dim, nstate>::update_maximum_local_wave_speed(DGBase<dim, double> &dg)
+{    
+    // Initialize the maximum local wave speed to zero
+    this->maximum_local_wave_speed = 0.0;
+
+    // Overintegrate the error to make sure there is not integration error in the error estimate
+    int overintegrate = 10;
+    dealii::QGauss<dim> quad_extra(dg.max_degree+1+overintegrate);
+    dealii::FEValues<dim,dim> fe_values_extra(*(dg.high_order_grid->mapping_fe_field), dg.fe_collection[dg.max_degree], quad_extra,
```

Comment:
@PranshulThakur does this also apply to `periodic_turbulence.cpp`?

Reply:
I had a look at `PeriodicTurbulence<dim, nstate>::compute_and_update_integrated_quantities(..)`  and there might be potential issues when the poly degree varies. For example, `n_dofs` in that function is set as a const before the cell loop begins [here](https://github.com/dougshidong/PHiLiP/blob/be278a2df9f38049b454227139aa415be5ed0288/src/flow_solver/flow_solver_cases/periodic_turbulence.cpp#L435), and values of `dg.solution` accessed [here](https://github.com/dougshidong/PHiLiP/blob/be278a2df9f38049b454227139aa415be5ed0288/src/flow_solver/flow_solver_cases/periodic_turbulence.cpp#L488) would mean that that solution coeffs of another cell (probably the cell containing the global index of 0 if cell->get_dof_indices() is setting the extra values to 0) will get used in the current cell. It might cause a bug if the code is run in serial and might segfault if it runs in parallel. I am not sure about how cell->get_dof_indices() works though and it's safer to set n_dofs w.r.t. cell->active_fe_index(). However, there shouldn't be any issues as long as the poly degree is same everywhere and is the same as max_degree.

Reply:
Ahh I remember this now, it was assumed that I would never be doing hp-adaptive simulations in periodic turbulence

Reply:
@PranshulThakur But the over-integration is only based on the local degree and hence you would just be over-integrating the local flux or error in this case.

Reply:
Yes, over-integration by using higher number of quadrature points but without changing the basis for solution space should be fine. However, in this case, FEValues (basis functions) is set to order `dg.max_degree` and would need greater number of coefficients than is available in `dg.solution`.